### PR TITLE
Show error popup when hint text gets type mismatch exception 

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
https://dimagi-dev.atlassian.net/browse/MOB-121
This PR will show popup dialog box with the XPathTypeMismatchException message while fetching question hint. 

cross-request: https://github.com/dimagi/commcare-core/pull/874

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

File: app/src/org/commcare/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -566,25 +566,32 @@ private boolean saveAnswersForCurrentScreen(boolean evaluateConstraints,
...
+                                        + index.getReference());
                     }
-                } else {
-                    Log.w(TAG,
-                            "Attempted to save an index referencing something other than a question: "
-                                    + index.getReference());
                 }
+            } catch (XPathException e) {
+                UserfacingErrorHandling.createErrorDialog(this, e.getLocalizedMessage(), true);
+            } catch (Exception e) {
```
Comment (shubham1g5) [inline app/src/org/commcare/activities/FormEntryActivity.java]: we only wanna catch XpathException here and let application crash in other failures. So we should remove the `Exception` catch from here and other places in PR where we have added it.

File: app/src/org/commcare/views/QuestionsView.java [hunk 2]
```diff
@@ -304,19 +306,26 @@ public SpannableStringBuilder getGroupLabel() {
...
+                    if (curHintText == null || !curHintText.equals(hintText)) {
+                        //If not, we can't do this trick
+                        hintText = null;
+                        break;
+                    }
                 }
             }
+        } catch (XPathException exception) {
+            UserfacingErrorHandling.createErrorDialog((CommCareActivity)getContext(), exception.getLocalizedMessage(), true);
+        } catch (Exception e) {
```
Comment (shubham1g5) [inline app/src/org/commcare/views/QuestionsView.java]: should remove the all Exception catch.

File: app/src/org/commcare/views/widgets/QuestionWidget.java [hunk 3]
```diff
@@ -502,7 +506,14 @@ private MediaLayout createHelpLayout() {
             text.setText(forceMarkdown(markdownText));
             text.setMovementMethod(LinkMovementMethod.getInstance());
         } else {
-            text.setText(mPrompt.getHelpText());
+            try {
+                text.setText(mPrompt.getHelpText());
+            } catch (XPathException exception) {
+                UserfacingErrorHandling.createErrorDialog((CommCareActivity)getContext(), exception.getLocalizedMessage(), true);
+            } catch (Exception e) {
```
Comment (shubham1g5) [inline app/src/org/commcare/views/widgets/QuestionWidget.java]: remove

File: app/src/org/commcare/views/widgets/QuestionWidget.java [hunk 4]
```diff
@@ -585,8 +596,15 @@ public void updateHintHeight(int maxHintHeight) {
...
      */
     private void addHintText() {
-        String s = mPrompt.getHintText();
-
+        String s = null;
+        try {
+            s = mPrompt.getHintText();
+        } catch (XPathException e) {
+            UserfacingErrorHandling.createErrorDialog((CommCareActivity)getContext(), e.getLocalizedMessage(), true);
+        } catch (Exception e) {
```
Comment (shubham1g5) [inline app/src/org/commcare/views/widgets/QuestionWidget.java]: remove

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Pull request merged**